packages/manager/apps/pci-databases-analytics/src/hooks/api/database/indexes/useDeleteIndex.hook.tsx

```diff
@@ -0,0 +1,39 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { deleteIndex, DeleteIndex } from '@/data/api/database/indexes.api';
+import { CdbError } from '@/data/api/database';
+
+export interface UseDeleteIndex {
+  onError: (cause: CdbError) => void;
+  onSuccess: () => void;
+}
+export function useDeleteIndex({
+  onError,
+  onSuccess: customOnSuccess,
+}: UseDeleteIndex) {
+  const queryClient = useQueryClient();
+  const mutation = useMutation({
+    mutationFn: (npInfo: DeleteIndex) => {
+      return deleteIndex(npInfo);
+    },
+    onError,
+    onSuccess: (_, variables) => {
```

Comment:
As a global improvement, I would not couple the query invalidation to the success callback, as it makes the hook not extendable or customizable.
By design, we should code here the success abstraction by letting the consumer (the view) call the right action related to the right purpose :)